Add Synapse 1.73 blog post (#1587)
matrix-org/matrix.org: gatsby/content/blog/2022/12/2022-12-07-synapse-1.73.mdx

---
date: '2022-12-07'
title: Synapse 1.73 released
categories:
- Releases
author: Mathieu Velten
---

And here is another update to your beloved Matrix server implementation, [Synapse
1.73](https://github.com/matrix-org/synapse/releases/tag/v1.73.0).

## Announcements

### Legacy Prometheus metric names removed

> As mentioned in previous blogpost, after disabling comes removing !

Reviewer: I think you accidentally dropped the "in" in the edit.

`enable_legacy_metrics` has been removed, sorry if it breaks your dashboard but

> the upgrade should not be that difficult, and you got some heads-up.

Reviewer 2: This sounds unnecessarily aggressive. Maybe start by giving a bit more context on what's going on (some people might be new to Matrix, and not everyone is following every announcement blog posts) and then explain objectively what's happening now, e.g.

> When releasing [Synapse 1.69](https://matrix.org/blog/2022/10/17/synapse-1-69-released#upcoming-removal-of-deprecated-prometheus-metric-names) a couple of months ago, we also announced the removal of old Prometheus metrics that have been replaced by more aptly named ones.
>
> Synapse 1.73 implements the final phase of this plan and entirely removes support for those metrics. As a result, the `enable_legacy_metrics` configuration option, which was introduced in [Synapse 1.71](https://matrix.org/blog/2022/11/22/synapse-1-72-released#announcements), has also been removed.
>
> Server administrators who are still relying on these legacy metric names are encouraged to update their dashboards at their earliest convenience. The list of renamed metrics can be found [here](https://matrix-org.github.io/synapse/v1.73/metrics-howto.html#renaming-of-metrics--deprecation-of-old-names-in-12), and more information can be found in the [upgrade notes](https://matrix-org.github.io/synapse/v1.73/upgrade#legacy-prometheus-metric-names-have-now-been-removed).

You can still find the full list of renamed metrics
[here](https://matrix-org.github.io/synapse/v1.73/metrics-howto.html#renaming-of-metrics--deprecation-of-old-names-in-12).

## The new stuff

### Performance

> A bunch of performance improvements have been included in this release.
> You can expect `/messages` to be [quite faster](https://github.com/matrix-org/synapse/pull/14527).

Reviewer 2: ```suggestion
A bunch of performance improvements have been included in this release, specifically around the [`/_matrix/client/v3/rooms/{roomId}/messages`](https://spec.matrix.org/v1.5/client-server-api/#get_matrixclientv3roomsroomidmessages) endpoint.
```

> You can expect `/messages` to be [quite faster](https://github.com/matrix-org/synapse/pull/14527).

Reviewer 3: We can add some graphs here: https://github.com/matrix-org/synapse/pull/14527

> Various optimizations have also been done on fetching aggregations of relationships.

Reviewer 3: We can link the PRs:

 - https://github.com/matrix-org/synapse/pull/14491
 - https://github.com/matrix-org/synapse/pull/14508
 - https://github.com/matrix-org/synapse/pull/14510

And add some graphs from: https://github.com/matrix-org/synapse/issues/13624#issuecomment-1323948527

### Extensible Events experimental support

Experimental support for Extensible Events has landed in Synapse.

> This is exciting since this global rework of events presentation has been in talk
> [for a while](https://github.com/matrix-org/matrix-spec-proposals/pull/1767), and we now have an
> implementation to experiment with that.
> 
> Beware that experimental is not a joke here, since the related MSCs hasn't even landed yet,
> and this has been implemented more to validate the approach than anything else.
> 
> If you are not a developer that wants to begin working on supporting them,
> please avoid enabling this since it will be useless to you and could be buggy.

Reviewer 2: ```suggestion
This is exciting since this global rework of events presentation has been in talk
[for a while](https://github.com/matrix-org/matrix-spec-proposals/pull/1767), and
having an implementation to experiment with is bringing the feature closer to completion.

Note that this support is still very much experimental as the related MSCs are still
under review and could change at any time, and therefore not recommended for use
in production.
```

a couple of months ago, we also announced the removal of old Prometheus metrics that have been replaced by more aptly named ones. 
he list of these metrics can be found [here](https://matrix-org.github.io/synapse/v1.73/metrics-howto.html#renaming-of-metrics--deprecation-of-old-names-in-12).

Synapse 1.73 implements the final phase of this plan and entirely removes support for those metrics. As a result,
the `enable_legacy_metrics` configuration option, which was introduced in
[Synapse 1.71](https://matrix.org/blog/2022/11/22/synapse-1-72-released#announcements), has also been removed.

Server administrators who are still relying on these legacy metric names are encouraged to update their dashboards at their earliest convenience.
For more information, please refer to the [upgrade notes](https://matrix-org.github.io/synapse/v1.73/upgrade#legacy-prometheus-metric-names-have-now-been-removed).

## The new stuff

### Performance

A bunch of performance improvements have been included in this release, specifically around
the [`/messages`](https://spec.matrix.org/v1.5/client-server-api/#get_matrixclientv3roomsroomidmessages) endpoint.

> [Improving `filter_events_for_client`](https://github.com/matrix-org/synapse/pull/14527) gave us a first nice bump as visible on this graph:

Reviewer 2: The graph is missing?

> [Improving `filter_events_for_client`](https://github.com/matrix-org/synapse/pull/14527) gave us a first nice bump as visible on this graph:

Reviewer 2: ```suggestion
[Improving `filter_events_for_client`](https://github.com/matrix-org/synapse/pull/14527) gave the matrix.org homeserver a first nice bump as visible on this graph:
```

Various optimizations around [fetching](https://github.com/matrix-org/synapse/pull/14491) [bundled](https://github.com/matrix-org/synapse/pull/14508)

> [aggregations](https://github.com/matrix-org/synapse/pull/14510) managed to got yet another nice improvement:

Reviewer 2: Better phrasing imo:

```suggestion
[aggregations](https://github.com/matrix-org/synapse/pull/14510) resulted in yet another nice improvement:
```

Also just a grammar nit on the original: "got" is past tense in the indicative, "to get" is infinitive, but "to got" does not exist.

> ![](/blog/img/2022-12-07-bundled-aggregations-improvement.png)
> 

Reviewer 2: Because apdexes can be confusing:

```suggestion
![](/blog/img/2022-12-07-bundled-aggregations-improvement.png)

Note that the graph from the first image, and the second graph from the second image are [apdexes](https://en.wikipedia.org/wiki/Apdex), which
is a measure that shows improvement when it goes up (as opposed to e.g. response times, which improve
when they go down).
```

### Extensible Events experimental support

Experimental support for Extensible Events has landed in Synapse.

This is exciting since this global rework of events presentation has been in talks
[for a while](https://github.com/matrix-org/matrix-spec-proposals/pull/1767), and
having an implementation to experiment with greatly helps bringing the feature
closer to completion.

Note that this support is still very much experimental as the related MSCs are still
under review and could change at any time, and therefore not recommended for use
in production.

## Everything else

> See the full [changelog](https://github.com/matrix-org/synapse/releases/tag/v1.73.0), for a
> complete list of changes in the release.

Reviewer 2: ```suggestion
See the full [changelog](https://github.com/matrix-org/synapse/releases/tag/v1.73.0), for a
complete list of changes in the release. Also please have a look at the [upgrade
notes](https://matrix-org.github.io/synapse/v1.73/upgrade#upgrading-to-v1730) for this version.
```

a couple of months ago, we also announced the removal of old Prometheus metrics that have been replaced by more aptly named ones. 
he list of these metrics can be found [here](https://matrix-org.github.io/synapse/v1.73/metrics-howto.html#renaming-of-metrics--deprecation-of-old-names-in-12).

Synapse 1.73 implements the final phase of this plan and entirely removes support for those metrics. As a result,
the `enable_legacy_metrics` configuration option, which was introduced in
[Synapse 1.71](https://matrix.org/blog/2022/11/22/synapse-1-72-released#announcements), has also been removed.

Server administrators who are still relying on these legacy metric names are encouraged to update their dashboards at their earliest convenience.
For more information, please refer to the [upgrade notes](https://matrix-org.github.io/synapse/v1.73/upgrade#legacy-prometheus-metric-names-have-now-been-removed).

## The new stuff

### Performance

A bunch of performance improvements have been included in this release, specifically around
the [`/messages`](https://spec.matrix.org/v1.5/client-server-api/#get_matrixclientv3roomsroomidmessages) endpoint.

> [Improving `filter_events_for_client`](https://github.com/matrix-org/synapse/pull/14527) gave the matrix.org homeserver a first nice bump as visible on this graph:

Reviewer 2: ```suggestion
[Improvements to event filtering on the client-server API](https://github.com/matrix-org/synapse/pull/14527) gave the matrix.org homeserver a first nice bump as visible on this graph:
```

`filter_events_for_client` means nothing to someone who's not a Synapse developer/contributor.

![](/blog/img/2022-12-07-apdex-filter-events-improvement.png)

Various optimizations around [fetching](https://github.com/matrix-org/synapse/pull/14491) [bundled](https://github.com/matrix-org/synapse/pull/14508)
[aggregations](https://github.com/matrix-org/synapse/pull/14510) resulted in yet another nice improvement:

> ![](/blog/img/2022-12-07-bundled-aggregations-improvement.png)

Reviewer 2: ```suggestion
[![](/blog/img/2022-12-07-bundled-aggregations-improvement.png)](/blog/img/2022-12-07-bundled-aggregations-improvement.png)
```

Linkifying the image because the labels are a bit hard to read on the preview.
(I've eyeballed the syntax, so hopefully this does the right think :sweat_smile:)
